### ISSUE TITLE
brew tests --no-compat fails

### ISSUE THREAD

**Issue by sjackman** [Linuxbrew/linuxbrew#730]
Ruby 1.9.3p551

See https://travis-ci.org/Linuxbrew/linuxbrew#L195-L203
```
==> brew tests --no-compat
/home/travis/build/Linuxbrew/linuxbrew/Library/Homebrew/test/testball.rb:1:in `<top (required)>': uninitialized constant Formula (NameError)
	from /home/travis/build/Linuxbrew/linuxbrew/Library/Homebrew/test/test_formula.rb:2:in `require'
	from /home/travis/build/Linuxbrew/linuxbrew/Library/Homebrew/test/test_formula.rb:2:in `<top (required)>'
	from /home/travis/build/Linuxbrew/linuxbrew/Library/Homebrew/test/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/rake_test_loader.rb:15:in `require'
	from /home/travis/build/Linuxbrew/linuxbrew/Library/Homebrew/test/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/rake_test_loader.rb:15:in `block in <main>'
	from /home/travis/build/Linuxbrew/linuxbrew/Library/Homebrew/test/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/rake_test_loader.rb:4:in `select'
	from /home/travis/build/Linuxbrew/linuxbrew/Library/Homebrew/test/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/rake_test_loader.rb:4:in `<main>'
rake aborted!
```

**Comment by sjackman** [comment]
@mikemcquaid Hi, Mike. `brew tests --no-compat` with Ruby 1.9.3p551 fails with the error message `testball.rb:1:in `<top (required)>': uninitialized constant Formula (NameError)`. Any idea why?

**Comment by mikemcquaid** [comment]
No idea, sorry. I'd recommend using Ruby 1.8 or Ruby 2.0 which are the versions we use.

**Comment by sjackman** [comment]
I haven't been able to reproduce this issue outside of Travis. Switch from Ruby 1.9 to Ruby 1.8 didn't resolve the issue. I haven't tested, but it seems like changing

https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/test/test_formula.rb#L2-L3
```
require "testing_env"
require "testball"
require "formula"
```
to
```
require "testing_env"
require "formula"
require "testball"
```

might resolve this (untested, but I'll give it a shot). `compat` imports `formula`

**Comment by sjackman** [comment]
@mikemcquaid This patch fixed the issue on Linuxbrew. I have no idea as to why it does not also affect Homebrew on Mac.
```diff
commit 4eab3f114df0ed95a7bf1e4fc7ae6db2e4e82f3a
Author: Shaun Jackman <sjackman@gmail.com>
Date:   Wed Jan 20 09:05:47 2016 -0800

    tests: Fix test_formula for Linuxbrew
    
    brew tests --no-compat on Travis fails with the error
    testball.rb:1:in `<top (required)>': uninitialized constant Formula (NameError)
    
    Fix Linuxbrew/linuxbrew#730

diff --git a/Library/Homebrew/test/test_formula.rb b/Library/Homebrew/test/test_formula.rb
index 8f372f6..5356bf5 100644
--- a/Library/Homebrew/test/test_formula.rb
+++ b/Library/Homebrew/test/test_formula.rb
@@ -1,6 +1,6 @@
 require "testing_env"
-require "testball"
 require "formula"
+require "testball"
 
 class FormulaTests < Homebrew::TestCase
   def test_formula_instantiation
```
